pkg/k8s/factory_functions.go

```diff
@@ -684,7 +686,7 @@ func ConvertToCiliumEndpoint(obj interface{}) interface{} {
 				Labels:      nil,
 				Annotations: nil,
 			},
-			Encryption: &concreteObj.Status.Encryption,
+			Encryption: &encryptVal,
```

Comment:
Is this change really needed?

Reply:
I don't understand it at the moment but its causing encryption key to be zero without it on my GKE cluster.